phoenix-core/src/main/java/org/apache/phoenix/schema/types/PDataType.java

```diff
@@ -1144,17 +1144,18 @@ public static PDataType fromLiteral(Object value) {
       return null;
     }
     for (PDataType type : PDataType.values()) {
-      if (type.isArrayType()) {
-        PhoenixArray arr = (PhoenixArray) value;
-        if ((type.getSqlType() == arr.baseType.sqlType + PDataType.ARRAY_TYPE_BASE)
-            && type.getJavaClass().isInstance(value)) {
-          return type;
-        }
-      } else {
-        if (type.getJavaClass().isInstance(value)) {
-          return type;
-        }
-      }
+			if (type.isArrayType() && type.getJavaClass().isInstance(value)) {
```

Comment:
If type.isArrayType() is true, then you should be able to cast it to a PhoenixArray. Is it because a PVarcharArray is storing the PJsonDataType instead of PVarchar? I think we'll need a PJsonArray.